app/upgrades.go

```diff
@@ -0,0 +1,122 @@
+package app
+
+import (
+	"context"
+	"fmt"
+
+	storetypes "cosmossdk.io/store/types"
+	upgradetypes "cosmossdk.io/x/upgrade/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+)
+
+const UpgradeName = "v10"
+
+func (app *WasmApp) RegisterUpgradeHandlers() {
+	upgradeInfo, err := app.UpgradeKeeper.ReadUpgradeInfoFromDisk()
+	if err != nil {
+		panic(fmt.Sprintf("failed to read upgrade info from disk %s", err))
+	}
+	app.Logger().Info("upgrade info", "name", upgradeInfo.Name, "height", upgradeInfo.Height)
+
+	if !app.UpgradeKeeper.IsSkipHeight(upgradeInfo.Height) {
+		app.Logger().Info("setting upgrade store loaders")
+		app.SetStoreLoader(upgradetypes.UpgradeStoreLoader(upgradeInfo.Height, &storetypes.StoreUpgrades{}))
+	}
+
+	// make this conditional for v10?
```

Comment:
yes this needs to be conditional, if someone runs the wrong image at the wrong height on an archive node they will be hosed.  This code is not really what I intended it should either follow the pattern from simapp or the one I had originally this is a cross between the two. 
simapp https://github.com/cosmos/cosmos-sdk/blob/v0.50.0/simapp/upgrades.go